packages/angular/src/app/components/theme-switcher/theme-switcher.component.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  Component, NgModule,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { DxButtonModule } from "devextreme-angular";
+
+import { currentTheme as currentVizTheme, refreshTheme } from 'devextreme/viz/themes';
+
+type Theme = 'dark'| 'light';
+
+const storageKey = 'themeViewer';
+const themeMarker = 'theme-';
+
+function getCurrentTheme(): Theme {
+  return window.localStorage[storageKey] || 'light';
+}
+
+function getThemeStyleSheets() {
+  return   [...(document.styleSheets as unknown as any[])]
+    .filter((styleSheet) => styleSheet?.href?.includes(themeMarker));
+}
+
+function switchTheme(themeName?: Theme) {
+  themeName = themeName || getCurrentTheme();
+
+  getThemeStyleSheets().forEach((styleSheet) => {
+      styleSheet.disabled = !styleSheet?.href?.includes(`${themeMarker}${themeName}`);
+    });
+
+  window.localStorage[storageKey] = themeName;
+
+  currentVizTheme(currentVizTheme().replace(/\.[a-z]+\.compact$/, `.${themeName}.compact`));
+  refreshTheme();
+}
+
+switchTheme();
```

Comment:
It seems that this line should be called where the application bootstrapped or in the root component's constructor.

Reply:
it works if switchTheme() is called only component